app/eventyay/plugins/sendmail/models.py

```diff
@@ -0,0 +1,396 @@
+    
+    :param positions: List of order position IDs.
+    :type positions: list[str]
+    
+    :param products: List of product IDs associated with this user.
+    :type products: list[str]
+    
+    :param team: Team ID if this is a team recipient.
+    :type team: int or None
+    
+    :param sent: Whether this recipient has been successfully sent the email.
+    :type sent: bool
+    
+    :param error: Error message if sending failed.
+    :type error: str or None
+    """
+    mail = models.ForeignKey(EmailQueue, on_delete=models.CASCADE, related_name="recipients")
+    email = models.EmailField()
+    orders = ArrayField(models.CharField(max_length=64), blank=True, default=list)
+    positions = ArrayField(models.CharField(max_length=64), blank=True, default=list)
+    products = ArrayField(models.CharField(max_length=64), blank=True, default=list)
+    team = models.IntegerField(null=True, blank=True)
+    sent = models.BooleanField(default=False)
+    error = models.TextField(null=True, blank=True)
+
+    class Meta:
+        unique_together = ("mail", "email")
+
+    def __str__(self):
+        return f"{self.email} for {self.mail_id}"
+
+
+class EmailQueueFilter(models.Model):
+    """
+    Stores structured filtering rules for recipient selection in EmailQueue.
+
+    :param mail: Associated EmailQueue.
+    :type mail: EmailQueue
+    
+    :param recipients: Target recipient scope: 'orders', 'attendees', or 'both'.
+    :type recipients: str
+    
+    :param sendto: Email roles or tags to include.
+    :type sendto: list[str]
+    
+    :param products: Filter by product IDs.
+    :type products: list[int]
+    
+    :param checkin_lists: Check-in list IDs to filter by.
+    :type checkin_lists: list[int]
+    
+    :param filter_checkins: Whether to filter based on check-in status.
+    :type filter_checkins: bool
+    
+    :param not_checked_in: Whether to include only recipients who haven’t checked in.
+    :type not_checked_in: bool
+    
+    :param subevent: Specific subevent ID to target.
+    :type subevent: int or None
+    
+    :param subevents_from: Filter subevents from this date/time onward.
+    :type subevents_from: datetime or None
+    
+    :param subevents_to: Filter subevents up to this date/time.
+    :type subevents_to: datetime or None
+    
+    :param created_from: Include orders created after this date/time.
+    :type created_from: datetime or None
+    
+    :param created_to: Include orders created before this date/time.
+    :type created_to: datetime or None
+    
+    :param orders: Explicit order IDs to include.
+    :type orders: list[int]
+    
+    :param teams: Team IDs to include (for team-based emails).
+    :type teams: list[int]
+    """
+    mail = models.OneToOneField(EmailQueue, on_delete=models.CASCADE, related_name="filters_data")
+
+    recipients = models.CharField(
+        max_length=10,
+        choices=[("orders", "Orders"), ("attendees", "Attendees"), ("both", "Both")],
```

Comment:
These choices should be defined as `enum` or `TextChoices`, so that it can be imported and used elsewhere.